src/components/PatchesPage/index.tsx

```diff
@@ -0,0 +1,168 @@
+    | { name: "Filter Commit Queue" }
+    | { name: "Change Page Size" }
+  >;
+  pageTitle: string;
+  patches?: PatchesPagePatchesFragment;
+  error?: ApolloError;
+}
+
+const PatchesPageCore: React.FC<Props> = ({
+  analyticsObject,
+  pageTitle,
+  patches,
+  error,
+}) => {
+  const bannersState = useBannerStateContext();
+  const dispatchBanner = useBannerDispatchContext();
+  const { replace } = useHistory();
+  const { search, pathname } = useLocation();
+  const { limit, page, includeCommitQueue } = getPatchesInputFromURLSearch(
+    search
+  );
+  const [
+    patchNameFilterValue,
+    patchNameFilterValueOnChange,
+  ] = useFilterInputChangeHandler(
+    MyPatchesQueryParams.PatchName,
+    false,
+    (filterBy: string) =>
+      analyticsObject?.sendEvent({ name: "Filter Patches", filterBy })
+  );
+  usePageTitle(pageTitle);
+  const onCheckboxChange = (): void => {
+    replace(
+      `${pathname}?${queryString.stringify(
+        {
+          ...queryString.parse(search, { arrayFormat }),
+          [MyPatchesQueryParams.CommitQueue]: !includeCommitQueue,
+        },
+        { arrayFormat }
+      )}`
+    );
+    // eslint-disable-next-line no-unused-expressions
+    analyticsObject?.sendEvent({ name: "Filter Commit Queue" });
+  };
+
+  return (
+    <PageWrapper>
+      <Banners
+        banners={bannersState}
+        removeBanner={dispatchBanner.removeBanner}
+      />
+      <PageTitle data-cy="patches-page-title">{pageTitle}</PageTitle>
+      <FiltersWrapperSpaceBetween>
+        <FlexRow>
+          <StyledInput
+            placeholder="Search Patch Descriptions"
+            onChange={patchNameFilterValueOnChange}
+            suffix={<Icon glyph="MagnifyingGlass" />}
+            value={patchNameFilterValue}
+            data-cy="patch-description-input"
+            width="25%"
+          />
+          <StatusSelector />
+        </FlexRow>
+        <StyledCheckbox
+          data-cy="commit-queue-checkbox"
+          onChange={onCheckboxChange}
+          label="Show Commit Queue"
+          checked={includeCommitQueue}
+        />
+      </FiltersWrapperSpaceBetween>
+      <PaginationRow>
+        <Pagination
+          pageSize={limit}
+          value={page}
+          totalResults={patches?.filteredPatchCount ?? 0}
+          dataTestId="my-patches-pagination"
+        />
+        <PageSizeSelector
+          dataTestId="my-patches-page-size-selector"
+          value={limit}
+          sendAnalyticsEvent={() =>
+            analyticsObject?.sendEvent({ name: "Change Page Size" })
+          }
+        />
+      </PaginationRow>
+      <ListArea patches={patches} error={error} />
+    </PageWrapper>
+  );
+};
+
+export const PatchesPage = withBannersContext(PatchesPageCore);
+
+const arrayFormat = "comma";
+export const getPatchesInputFromURLSearch = (
+  search: string
+): UserPatchesQueryVariables["patchesInput"] => {
+  const parsed = queryString.parse(search, { arrayFormat });
```

Comment:
We have a util function for this https://github.com/evergreen-ci/spruce/blob/ef259329ed600538751623375ed2fb0405697586/src/utils/queryString/parseQueryString.ts#L3